python/tester/upload.py

```diff
@@ -46,31 +42,32 @@ class Instrument:
 _log.setLevel(logging.INFO)
 
 
-def process_group(producer, visit_infos, uploader):
+def process_group(kafka_url, visit_infos, uploader):
     """Simulate the observation of a single on-sky pointing.
 
     Parameters
     ----------
-    producer : `confluent_kafka.Producer`
-        The client that posts ``next_visit`` messages.
-    visit_infos : `set` [`activator.Visit`]
+    kafka_url : `str`
+        The URL of the Kafka REST Proxy to send ``next_visit`` messages to.
+    visit_infos : `set` [`activator.FannedOutVisit`]
```

Comment:
Why use `FannedOutVisit` here? I would have thought the uploaders could just use `BareVisit` internally.

Reply:
Not really because how `raw_pool` is unless we rewrite it completely.  The thing is that we still want to keep track of files at the detector level and hence `FannedOutVisit` is quite convenient for it.  If we use `BareVisit` we'd still need those detector IDs somehow. It seems doable but I'm not sure if that'd really be better.